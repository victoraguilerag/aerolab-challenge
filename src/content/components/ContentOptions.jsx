import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'
import '../../stylesheet/responsive-stylesheet.css'
import Filter from './Filter.jsx'
import previusIcon from '../../icons/arrow-left.svg'
import nextIcon from '../../icons/arrow-right.svg'

function ContentOptions (props) {
  const {
    prev,
    next,
    handlePreviousPage,
    handleNextPage,
    pageLength,
    productsLength,
    filters,
    activeFilter,
    handleFilterChange
  } = props
  return (
    <div className="content-options" style={styles.contentOptions}>
      <div className="options" style={styles.options}>
        <div className="products-count-options" style={styles.productsCount}>
          {`${pageLength} of ${productsLength} items`}
        </div>
        <div className="divider" style={styles.divider} />
        <div className="filters" style={styles.filtersContainer}>
          <p className="filters-label" style={styles.filtersLabel}>
            Sort by:
          </p>
          <ul className="filters" style={styles.filters}>
            {
              filters.map(
                filter => <Filter key={filter} active={(filter === activeFilter)} handleFilterChange={handleFilterChange} label={filter} />
              )
            }
          </ul>
        </div>
      </div>
      <div className="controls" style={styles.controls}>
        {
          prev &&
          <img
            className="previus"
            src={previusIcon}
            style={styles.control}
            width="46px"
            height="46px"
            alt="previus"
            onClick={handlePreviousPage}
          />
        }
        {
          next &&
          <img
            className="next"
            src={nextIcon}
            style={styles.control}
            width="46px"
            height="46px"
            alt="next"
            onClick={handleNextPage}
          />
        }
      </div>
    </div>
  )
}

const styles = {
  'contentOptions': {
    'display': 'flex',
    'flexFlow': 'column wrap',
    'justifyContent': 'flex-start',
    'alignItems': 'flex-start',
    'margin': '0 auto',
    'maxWidth': '1176px',
    'height': '70px',
    'borderBottom': 'solid 1px #d9d9d9'
  },
  'options': {
    'display': 'flex'
  },
  'productsCount': {
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left'
  },
  'divider': {
    'background': '#d9d9d9',
    'width': '1px',
    'height': '49px',
    'margin': '5px',
    'marginLeft': '23px'
  },
  'filtersContainer': {
    'display': 'flex'
  },
  'filtersLabel': {
    'userSelect': 'none',
    'fontSize': '24px',
    'color': '#a3a3a3',
    'letterSpacing': '-0.15px',
    'textAlign': 'left',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'margin': '0',
    'marginLeft': '24px'
  },
  'filters': {
    'listStyle': 'none',
    'display': 'flex',
    'flexFlow': 'row',
    'alignItems': 'center',
    'margin': '0',
    'padding': '0'
  },
  'controls': {
    'alignSelf': 'flex-end',
    'height': '100%',
    'display': 'flex',
    'alignItems': 'flex-start'
  },
  'control': {
    'userSelect': 'none',
    'margin': '0 5px',
    'cursor': 'pointer'
  }
}

ContentOptions.propTypes = {
  prev: PropTypes.bool.isRequired,
  next: PropTypes.bool.isRequired,
  handlePreviousPage: PropTypes.func.isRequired,
  handleNextPage: PropTypes.func.isRequired,
  pageLength: PropTypes.number.isRequired,
  productsLength: PropTypes.number.isRequired,
  filters: PropTypes.instanceOf(Immutable.List).isRequired,
  activeFilter: PropTypes.string.isRequired,
  handleFilterChange: PropTypes.func.isRequired
}

export default ContentOptions
