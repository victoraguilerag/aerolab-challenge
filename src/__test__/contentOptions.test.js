import React from 'react'
import ContentOptions from '../content/components/ContentOptions.jsx'
import renderer from 'react-test-renderer'
import { fromJS } from 'immutable'

const filters = fromJS(['Most recent', 'Lowest price', 'Highest price'])
const functionMock = () => console.log('mock')
const pageLength = 2
const productsLength = 16

test('Content Options renders', () => {
  const component = renderer.create(
    <ContentOptions
      filters={filters}
      prev={false}
      next={true}
      handlePreviousPage={functionMock}
      handleNextPage={functionMock}
      pageLength={pageLength}
      productsLength={productsLength}
      handleFilterChange={functionMock}
      activeFilter="active filter"
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
