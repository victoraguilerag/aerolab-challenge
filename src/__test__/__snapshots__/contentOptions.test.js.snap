// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Options renders 1`] = `
<div
  className="content-options"
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottom": "solid 1px #d9d9d9",
      "display": "flex",
      "flexFlow": "column wrap",
      "height": "70px",
      "justifyContent": "flex-start",
      "margin": "0 auto",
      "maxWidth": "1176px",
    }
  }
>
  <div
    className="options"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="products-count-options"
      style={
        Object {
          "alignItems": "center",
          "color": "#616161",
          "display": "flex",
          "fontSize": "24px",
          "justifyContent": "center",
          "letterSpacing": "-0.15px",
          "lineHeight": "48px",
          "textAlign": "left",
        }
      }
    >
      2 of 16 items
    </div>
    <div
      className="divider"
      style={
        Object {
          "background": "#d9d9d9",
          "height": "49px",
          "margin": "5px",
          "marginLeft": "23px",
          "width": "1px",
        }
      }
    />
    <div
      className="filters"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <p
        className="filters-label"
        style={
          Object {
            "alignItems": "center",
            "color": "#a3a3a3",
            "display": "flex",
            "fontSize": "24px",
            "justifyContent": "center",
            "letterSpacing": "-0.15px",
            "margin": "0",
            "marginLeft": "24px",
            "textAlign": "left",
            "userSelect": "none",
          }
        }
      >
        Sort by:
      </p>
      <ul
        className="filters"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexFlow": "row",
            "listStyle": "none",
            "margin": "0",
            "padding": "0",
          }
        }
      >
        <li
          className="filter"
          onClick={[Function]}
          style={
            Object {
              "background": "#ededed",
              "borderRadius": "100px",
              "cursor": "pointer",
              "height": "48px",
              "margin": "0 5px",
              "marginLeft": "2%",
              "userSelect": "none",
              "width": "163px",
            }
          }
        >
          <p
            className="filter-text"
            style={
              Object {
                "alignItems": "center",
                "color": "#a3a3a3",
                "display": "flex",
                "fontSize": "1em",
                "height": "100%",
                "justifyContent": "center",
                "margin": "0",
                "padding": "0 10px",
                "textAlign": "center",
                "userSelect": "none",
              }
            }
          >
            Most recent
          </p>
        </li>
        <li
          className="filter"
          onClick={[Function]}
          style={
            Object {
              "background": "#ededed",
              "borderRadius": "100px",
              "cursor": "pointer",
              "height": "48px",
              "margin": "0 5px",
              "marginLeft": "2%",
              "userSelect": "none",
              "width": "163px",
            }
          }
        >
          <p
            className="filter-text"
            style={
              Object {
                "alignItems": "center",
                "color": "#a3a3a3",
                "display": "flex",
                "fontSize": "1em",
                "height": "100%",
                "justifyContent": "center",
                "margin": "0",
                "padding": "0 10px",
                "textAlign": "center",
                "userSelect": "none",
              }
            }
          >
            Lowest price
          </p>
        </li>
        <li
          className="filter"
          onClick={[Function]}
          style={
            Object {
              "background": "#ededed",
              "borderRadius": "100px",
              "cursor": "pointer",
              "height": "48px",
              "margin": "0 5px",
              "marginLeft": "2%",
              "userSelect": "none",
              "width": "163px",
            }
          }
        >
          <p
            className="filter-text"
            style={
              Object {
                "alignItems": "center",
                "color": "#a3a3a3",
                "display": "flex",
                "fontSize": "1em",
                "height": "100%",
                "justifyContent": "center",
                "margin": "0",
                "padding": "0 10px",
                "textAlign": "center",
                "userSelect": "none",
              }
            }
          >
            Highest price
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="controls"
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "flex-end",
        "display": "flex",
        "height": "100%",
      }
    }
  >
    <img
      alt="next"
      className="next"
      height="46px"
      onClick={[Function]}
      src={Object {}}
      style={
        Object {
          "cursor": "pointer",
          "margin": "0 5px",
          "userSelect": "none",
        }
      }
      width="46px"
    />
  </div>
</div>
`;
